### PR TITLE
nvme_smart_log_test: call smart log for all ns based on LPA 

### PR DESCRIPTION
**Pull request by Arunpandian15**
bug fix - added check condition in nvme_smart_log_test.py to check smart log support per namespace
and call accordingly.
added supp_check_id_ctrl wrapper in nvme_test.py to check support
based on passed key value to make it generic.

Signed-off-by: Arunpandian J <apj.arun@samsung.com>

### CODE REVIEW

**Comment by igaw** [comment]
The commits are in the wrong order. Please introduce first the function and then the callside.

File: tests/nvme_smart_log_test.py [hunk 1]
```diff
@@ -83,4 +83,6 @@ def get_smart_log_all_ns(self):
     def test_smart_log(self):
         """ Testcase main """
         assert_equal(self.get_smart_log_ctrl(), 0)
-        assert_equal(self.get_smart_log_all_ns(), 0)
+        smlp = self.supp_check_id_ctrl("lpa")
```
Comment (igaw) [inline tests/nvme_smart_log_test.py:86]: Please introduce supp_check_id_ctrl first in the patch series.

**Review by igaw — CHANGES_REQUESTED** [review]
reorder the patches please

File: tests/nvme_smart_log_test.py (hunk 1 above)
Comment (Arunpandian15) [reply tests/nvme_smart_log_test.py:86]: Reorder the patch series as per the ask. Thanks @igaw

**Comment by Arunpandian15** [comment]
> reorder the patches please

Reordered the patches. Thanks @igaw

**Comment by igaw** [comment]
Thanks!

**Pull request merged**